File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.js
function solution(s) {
    let zero = 0;
    let loop = 0;

    while (s !== '1') {
        let lengthBefore = s.length;

        s = s.replace(/0/g, '');

        let lengthAfter = s.length;
        zero += lengthBefore - lengthAfter;

        s = lengthAfter.toString(2);
        loop++;
    }

    return [loop, zero];
}

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.py
def solution(s):
    value = s
    answer = [0, 0]
    
    while value != '1':
        first = len(value)
        value = value.replace('0','');
        target = len(value)
        answer[1] += first - target

        value = str(bin(target)[2:] )
        answer[0] += 1    
    
    return answer